src/tools/store.cpp

```diff
@@ -204,61 +196,83 @@ bool GenerateDataStoreOptions(const int argc,
+        throw util::exception("valid .nodes file must be specified");
     }
 
     path_iterator = paths.find("edgesdata");
     if (path_iterator == paths.end() || path_iterator->second.string().empty() ||
         !boost::filesystem::is_regular_file(path_iterator->second))
     {
-        throw exception("valid .edges file must be specified");
+        throw util::exception("valid .edges file must be specified");
     }
 
     path_iterator = paths.find("geometry");
     if (path_iterator == paths.end() || path_iterator->second.string().empty() ||
         !boost::filesystem::is_regular_file(path_iterator->second))
     {
-        throw exception("valid .geometry file must be specified");
+        throw util::exception("valid .geometry file must be specified");
     }
 
     path_iterator = paths.find("ramindex");
     if (path_iterator == paths.end() || path_iterator->second.string().empty() ||
         !boost::filesystem::is_regular_file(path_iterator->second))
     {
-        throw exception("valid .ramindex file must be specified");
+        throw util::exception("valid .ramindex file must be specified");
     }
 
     path_iterator = paths.find("fileindex");
     if (path_iterator == paths.end() || path_iterator->second.string().empty() ||
         !boost::filesystem::is_regular_file(path_iterator->second))
     {
-        throw exception("valid .fileindex file must be specified");
+        throw util::exception("valid .fileindex file must be specified");
     }
 
     path_iterator = paths.find("namesdata");
     if (path_iterator == paths.end() || path_iterator->second.string().empty() ||
         !boost::filesystem::is_regular_file(path_iterator->second))
     {
-        throw exception("valid .names file must be specified");
+        throw util::exception("valid .names file must be specified");
     }
 
     path_iterator = paths.find("timestamp");
     if (path_iterator == paths.end() || path_iterator->second.string().empty() ||
         !boost::filesystem::is_regular_file(path_iterator->second))
     {
-        throw exception("valid .timestamp file must be specified");
+        throw util::exception("valid .timestamp file must be specified");
     }
 
     return true;
 }
+
+int main(const int argc, const char *argv[]) try
+{
+    util::LogPolicy::GetInstance().Unmute();
+
+    storage::DataPaths paths;
+    if (!generateDataStoreOptions(argc, argv, paths))
+    {
+        return EXIT_SUCCESS;
+    }
+
+    storage::Storage storage(paths);
+    return storage.Run();
+}
+catch (const std::bad_alloc &e)
+{
+    util::SimpleLogger().Write(logWARNING) << "[exception] " << e.what();
+    util::SimpleLogger().Write(logWARNING)
+        << "Please provide more memory or disable locking the virtual "
+           "address space (note: this makes OSRM swap, i.e. slow)";
+    return EXIT_FAILURE;
 }
+catch (const std::exception &e)
+{
+    util::SimpleLogger().Write(logWARNING) << "caught exception: " << e.what();
```

Comment:
`return EXIT_FAILURE` here, too --- and for all tools, check with

    ag 'catch \(' -A 5 src/tools